src/shared/components/dataAccessTokens/DataAccessTokens.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from "react";
+import AppConfig from "appConfig";
+import {If, Then, Else} from 'react-if';
+import client from "shared/api/cbioportalClientInstance";
+import {AppStore} from "../../../AppStore";
+import LoadingIndicator from "../loadingIndicator/LoadingIndicator";
+import {observer} from "mobx-react";
+import {observable} from "mobx";
+import {buildCBioPortalPageUrl} from "../../api/urls";
+import {isNullOrUndefined} from "util";
+import fileDownload from 'react-file-download';
+
+export class UserDataAccessToken {
+    @observable token : string;
+    @observable creationDate: string;
+    @observable expirationDate: string;
+    @observable username: string;
+    constructor(token: string, creationDate: string, expirationDate: string, username: string) {
+        this.token = token;
+        this.creationDate = creationDate;
+        this.expirationDate = expirationDate;
+        this.username = username;
+    }
+}
+
+export interface IDataAccessTokensProps {
+    token?:string;
+    creationDate?:string;
+    expirationDate?:string;
+    loadingComponent?:JSX.Element;
+    appStore:AppStore;
+}
+
+function buildDataAccessTokenFileContents(dat:UserDataAccessToken | undefined) {
+    if (!isNullOrUndefined(dat)) {
+        const fileContents = "token: " + dat!.token + "\n" +
+            "creation_date: " + dat!.creationDate + "\n" +
+            "expiration_date: " + dat!.expirationDate;
+        return fileContents;
+    } else {
+        alert("Cannot create Data Access Token file for user with non-existent tokens.");
+        return null;
+    }
+}
+
+@observer
+export class DataAccessTokens extends React.Component<IDataAccessTokensProps, {}> {
+    public static defaultProps: Partial<IDataAccessTokensProps> = {
+        loadingComponent:<LoadingIndicator isLoading={true}/>
+    };
+
+    constructor(props: IDataAccessTokensProps) {
+        super(props);
+    }
+
+    getDatDropdownList() {
+        const listItems = [
+            {
+                id:"signout",
+                action:<a href={buildCBioPortalPageUrl(this.props.appStore.logoutUrl)}>Sign out</a>,
+                hide:false
+            },
+            {
+                id:"datDownload",
+                action:<a onClick={() => this.downloadDataAccessTokenFile()}>Download token</a>,
+                hide:AppConfig.serverConfig.authenticationMethod === "social_auth"
```

Comment:
This covers our main use cases. For future enhancement, the backend API for DAT could be probed to see whether the individual user has permission to use data access tokens on that portal. Then the menu item could be disabled/hidden for individual users who are prohibited from using tokens. I see that you have dropped the Store for holding state for the user's tokens, which makes sense. But we may want to revive that soon to hold user state information ... such as whether they have permissions to use data access tokens, and maybe other user-specific state such as available sessions they have saved through the session-service.